homeassistant/components/binary_sensor/mystrom.py

```diff
@@ -0,0 +1,98 @@
+"""
+Support for the myStrom buttons.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.mystrom/
+"""
+import asyncio
+import logging
+
+from homeassistant.const import HTTP_UNPROCESSABLE_ENTITY
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.components.binary_sensor import (BinarySensorDevice, DOMAIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['http']
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up myStrom Binary Sensor."""
+    button = None
+
+    hass.http.register_view(MyStromView(button, async_add_devices))
+
+    return True
+
+
+class MyStromView(HomeAssistantView):
+    """View to handle requests from myStrom buttons."""
+
+    url = '/api/mystrom'
+    name = 'api:mystrom'
+
+    def __init__(self, button, add_devices):
+        """Initialize the myStrom URL endpoint."""
+        self.button = button
+        self.add_devices = add_devices
+
+    @asyncio.coroutine
+    def get(self, request):
+        """The GET request received from a myStrom button."""
+        res = yield from self._handle(request.app['hass'], request.GET)
+        return res
+
+    @asyncio.coroutine
+    def _handle(self, hass, data):
+        """Handle requests to the myStrom endpoint."""
+        button_action = list(data.keys())[0]
+        button_id = data[button_action]
+        entity_id = '{}.{}_{}'.format(DOMAIN, button_id, button_action)
+        state = hass.states.get(entity_id)
```

Comment:
We should not interact directly with `hass` but instead use the abstract base class for binary sensor.